deployment/layoutCmd.py

```diff
@@ -0,0 +1,84 @@
+import os
+from kubernetes import client, config
+import yaml
+
+
+def generate_layout(output_file):
+    # init client
+    config.load_kube_config()
+    v1 = client.CoreV1Api()
+
+    # api server url
+    api_servers_url = v1.api_client.configuration.host
+    # generate dashboard-url
+    services = v1.list_service_for_all_namespaces(field_selector="metadata.name=kubernetes-dashboard", pretty=False, timeout_seconds=56, watch=False)
+    dashboard_service = services.items[0]
+    dashboard_url = "http://{0}:80".format(dashboard_service.spec.cluster_ip)
+
+    # query k8s nodes
+    nodes = v1.list_node(pretty=False, timeout_seconds=56, watch=False)
+    addressesList = map(lambda node: node.status.addresses, nodes.items)
+    machineList = []
+    for addresses in addressesList:
+        machine = dict()
+        machine['machine-type'] = 'GENERIC'
+        for address in addresses:
+            if address.type == 'InternalIP':
+                machine['hostip'] = address.address
+            if address.type == 'Hostname':
+                machine['hostname'] = address.address
+                # TODO nodename == hostname on aks
+                machine['nodename'] = address.address
+        machineList.append(machine)
+    machineList.sort(key=lambda k: k['hostname'])
+
+    # assgin pai-master
+    master = machineList[0]
+    master['pai-master'] = 'true'
+    master['zkid'] = 1
+
+    # assign pai-workers
+    workers = machineList[1:] if len(machineList) > 1 else machineList
+    for worker in workers:
+        worker['pai-worker'] = 'true'
+
+    # the default sku
+    machineSku = yaml.load("""
+GENERIC:
```

Comment:
Please directly init it as a dict

Reply:
Yeah, implement this way for readability.
So I can easily got what the yaml would be.

Reply:
I don't think so . Init it from a dict will also be readability

---
In reply to: [252225274](https://github.com/Microsoft/pai/pull/1980#discussion_r252225274) [](ancestors = 252225274)